gamehub-backend/src/main/java/com/example/gamehubbackend/services/UserService.java

```diff
@@ -17,23 +20,54 @@ public class UserService {
 
     private final UserRepository userRepository;
     private final IdService idService;
+    private final PasswordEncoder passwordEncoder;
 
     public List<User> getAllUser() {
         return userRepository.findAll();
     }
 
+    public UserResponse getLoggedInUser() {
+        var principal = (org.springframework.security.core.userdetails.User) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        return getUserByUsername(principal.getUsername());
+    }
+
     public User getUserById(String id) {
         return userRepository.findById(id).orElseThrow(() -> new UserNotFoundException("No user found with id: " + id));
     }
 
-    public User getUserByGitHubId(String githubId) {
-        return userRepository.findByGitHubId(githubId).orElseThrow(() -> new UserNotFoundException("No user found with gitHubId: " + githubId));
+    public UserResponse getUserByGitHubId(String githubId) {
+        User user = userRepository.findByGitHubId(githubId).orElseThrow(() -> new UserNotFoundException("No user found with gitHubId: " + githubId));
+
+        return new UserResponse(
+                user.id(),
+                user.gitHubId(),
+                user.username(),
+                user.avatarUrl(),
+                user.role(),
+                user.gameLibrary()
+        );
+    }
+
+    public UserResponse getUserByUsername(String username) {
+        User user = userRepository.findByUsername(username).orElseThrow(() -> new UserNotFoundException("No user found with username: " + username));
```

Comment:
don't think it matters, but would expect UsernameNotFoundException here
```suggestion
        User user = userRepository.findByUsername(username).orElseThrow(() -> new UsernameNotFoundException("No user found with username: " + username));
```